### PR TITLE
solarish adding ucontext struct

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: src/unix/solarish/x86_64.rs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        pub status: u32,
+        pub xstatus: u32,
+    }
+}
+
+s_no_extra_traits! {
+    #[cfg(libc_union)]
+    pub union __c_anonymous_fp_reg_set {
+        pub fpchip_state: __c_anonymous_fpchip_state,
+        pub f_fpregs: [[u32; 13]; 10],
```
Comment (Amanieu) [inline src/unix/solarish/x86_64.rs:26]: This really should be `[u32; 130]` but that may not work with older Rust versions used in CI.

@JohnTitor Thoughts?

Comment (devnexen) [reply src/unix/solarish/x86_64.rs:26]: yes why I did just for precaution

Comment (JohnTitor) [reply src/unix/solarish/x86_64.rs:26]: Yes, we've been using this workaround in some places and it should work fine.

**Pull request merged**